src/main/java/org/apache/rocketmq/flink/source/reader/RocketMQPartitionSplitReader.java

```diff
@@ -148,30 +153,44 @@ public RecordsWithSplitIds<Tuple3<T, Long, Long>> fetch() throws IOException {
                     if (wakeup) {
                         LOG.info(
                                 String.format(
-                                        "Wake up pulling messages of topic[%s] broker[%s] queue[%d] tag[%s] from offset[%d].",
+                                        "Wake up pulling messages of topic[%s] broker[%s] queue[%d] tag[%s] sql[%s] from offset[%d].",
                                         messageQueue.getTopic(),
                                         messageQueue.getBrokerName(),
                                         messageQueue.getQueueId(),
                                         tag,
+                                        sql,
                                         messageOffset));
                         wakeup = false;
                         recordsBySplits.prepareForRead();
                         return recordsBySplits;
                     }
-                    pullResult =
-                            consumer.pull(
-                                    messageQueue, tag, messageOffset, MAX_MESSAGE_NUMBER_PER_BLOCK);
+                    if (StringUtils.isNotEmpty(tag)) {
+                        pullResult =
+                                consumer.pull(
+                                        messageQueue,
+                                        tag,
+                                        messageOffset,
+                                        MAX_MESSAGE_NUMBER_PER_BLOCK);
+                    } else {
```

Comment:
Should this check whether `sql` is empty?

Reply:
> Should this check whether `sql` is empty?

@SteNicholas I have add this check condition.Please cc